### PR TITLE
[FEATURE] allow install via composer

### PR DESCRIPTION
**Pull request by kaystrobach**

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -1,8 +1,17 @@
 {
+    "name": "nextcloud/server",
+    "description": "Media gallery for ownCloud which includes previews for all media types supported by your ownCloud installation.",
```
Comment (nickvergessen) [inline composer.json]: *cough* no?

**Review by nickvergessen — CHANGES_REQUESTED** [review]

File: composer.json (hunk 1 above)
Comment (MariusBluem) [inline composer.json]: Replace it with: 

_A safe home for all your data. Access, share and protect your files, calendars, contacts, communication & more at home and in your enterprise._

**Comment by kaystrobach** [comment]
changed as requested

**Comment by rullzer** [comment]
This will fail hard I bet because the 3rdparty repo is a submodule.

**Comment by kaystrobach** [comment]
I will try to get this fixed afterwards.

please see https://github.com/owncloud/core/issues/11973#issuecomment-74882008 for the same discussion in owncloud, i think it's pretty crucial to have full composer support in a midterm.
So security of libs can be easily checked with tools like:

See http://stackoverflow.com/questions/18100420/how-do-i-use-a-git-submodule-with-a-composer-loaded-library for details ... should work with `prefer-dist` option set.

**Comment by nickvergessen** [comment]
> So security of libs can be easily checked with tools like:

Our libs are already handled by composer:
https://github.com/nextcloud/3rdparty/blob/master/composer.json

**Comment by kaystrobach** [comment]
would be awesome to have that in the main composer.json to allow fast in reliable install via composer, this would also allow professional extension installations (with dependencies)

**Comment by LukasReschke** [comment]
> please see owncloud/core#11973 (comment) for the same discussion in owncloud, i think it's pretty crucial to have full composer support in a midterm.

We won't go with a full composer alike approach anytime soon from my opinion as this moves libraries out of our control (e.g. random person can delete their libs) plus I yet fail to see some security features such as signing the library. I am not willed to accept situations where a third-party dev can randomly change the code that is offered to download for stability and security reasons.

But we do check the security of the libraries already using VersionEye for example.

**Comment by kaystrobach** [comment]
How do you deploy nextxloud then?

Fork repo put submodules for each plugin in, use dotenv for conf?

**Comment by kaystrobach** [comment]
Feels a bit strange to have more than composer.json + composer.lock in the dist forlder for deployment 😉

**Comment by LukasReschke** [comment]
> How do you deploy nextxloud then?
> Fork repo put submodules for each plugin in, use dotenv for conf?

Download the ZIP or TAR from our webpage as officially documented 😉  – If one wanna use git, well, then one has to checkout whatever submodules an app may ship and figure that out for themselves.

**Comment by rullzer** [comment]
Also there is the matter of not everything that is in the tarball being in the main repo. So you endup with a lot of git commands anyway.

So lets close this as nothing is happening. For installation please use the ZIP/TAR ball and if you want to use git then write a script that does all the additional checkouts.

**Pull request closed without merging**